include/circt/Dialect/SSP/SSPOps.td

```diff
@@ -24,25 +24,30 @@ def InstanceOp : SSPOp<"instance",
   let description = [{
     This operation represents an instance of a static scheduling problem,
     comprised of an operator library (`OperatorLibraryOp`, a container for
-    `OperatorTypeOp`s) and the dependence graph (`DependenceGraphOp`, a
+    `OperatorTypeOp`s), an resource library (`ResourceLibraryOp`, a container
+    for `ResourceTypeOp`s), and the dependence graph (`DependenceGraphOp`, a
     container for `OperationOp`s). The instance and its components (operations,
-    operator types and dependences) can carry properties, i.e. special MLIR
-    attributes inheriting from the TableGen classes in `PropertyBase.td`. The
-    `ssp` dialect provides attribute definitions (and short-form
-    pretty-printing) for CIRCT's built-in scheduling problems.
+    operator types, resource types and dependences) can carry properties,
+    i.e. special MLIR attributes inheriting from the TableGen classes in
+    `PropertyBase.td`. The `ssp` dialect provides attribute definitions (and
+    short-form pretty-printing) for CIRCT's built-in scheduling problems.
     
     **Example**
     ```mlir
     ssp.instance @canis14_fig2 of "ModuloProblem" [II<3>] {
       library {
-        operator_type @MemPort [latency<1>, limit<1>]
+        operator_type @Memory [latency<1>]
         operator_type @Add [latency<1>]
       }
+      resource {
+        resource_type @ReadPort [limit<1>]
+        resource_type @WritePort [limit<1>]
+      }
       graph {
-        %0 = operation<@MemPort> @load_A(@store_A [dist<1>]) [t<2>]
-        %1 = operation<@MemPort> @load_B() [t<0>]
-        %2 = operation<@Add> @add(%0, %1) [t<3>]
-        operation<@MemPort> @store_A(%2) [t<4>]
+        %0 = operation<@Memory> @load_A uses[@ReadPort](@store_A [dist<1>]) [t<2>]
+        %1 = operation<@Memory> @load_B uses[@ReadPort]() [t<0>]
+        %2 = operation<@Add> @add(%0, %1) [t<3>]  // no `resource_type` needed
+        operation<@Memory> @store_A uses[@WritePort](%2) [t<4>]
```

Comment:
I think the "uses" syntax should go after the arguments. So something like: `operation<@Memory> @store_A(%2) uses[@WritePort] [t<4>]`